src/app_service/service/saved_address/service.nim

```diff
@@ -56,13 +72,50 @@ proc createOrUpdateSavedAddress*(self: Service, name, address: string): string =
 
 proc deleteSavedAddress*(self: Service, address: string): string =
   try:
-    let response = backend.deleteSavedAddress(address)
+    var response = backend.deleteSavedAddress(address)
+    if any(self.favouriteAddresses, proc (x: FavouriteDto): bool = x.address == address):
+      discard backend.deleteFavourite(address)
+
+    if not response.error.isNil:
+      raise newException(Exception, response.error.message)
+
+    self.updateAddresses()
+    return ""
+  except Exception as e:
+    let errDesription = e.msg
+    return errDesription
+
+proc setFavouriteAddressState*(self: Service, address: string, favourite: bool): string =
```

Comment:
@stefandunca to me this name is somehow misleading... when you read it, it says `setFavouriteAddressState` and I expect that is should set something, but in the body I see that it add/delete address. Wdyt of renaming it to `proc addRemoveFavouriteAddress*(self: Service, address: string, add: bool)` or any other more intuitive option (an ideal case would be to have 2 procs, `addToFavourite(self: Service, address: string)` and `removeFromFavourite(self: Service, address: string)`?

Reply:
You are right. It seems I moved the presentation layer abstraction to the service layer. Fixed it, please have another look